### PR TITLE
visitorckw proposal: Runtime Profile-Guided Optimization Sorting Library

### PR DESCRIPTION
**Pull request by visitorckw**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [x] passing tests/CI

### CODE REVIEW

File: project/visitorckw/README.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+
+## Schedule
+
+Planning phase (6 weeks from 9/18 to 10/30): Think up presentation, then set up the environment
+and get familiar with the algorithm
+
+Week 1 (10/30):  Project Initiation
+
+- Define project objectives, scope, and requirements.
```
Comment (yungyuc) [inline project/visitorckw/README.md:154]: It's fine to keep it here, but I'd like to remind that the definition of a project is usually done before the implementation.

File: project/visitorckw/README.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+- Set up the development environment and tools.
+- Create a project plan, including milestones and deliverables.
+
+Week 2 (11/6): Data Structures and API Design
+
+- Design the data structures for arrays and linked lists.
```
Comment (yungyuc) [inline project/visitorckw/README.md:160]: Do you have a plan for memory management?  Array and linked list use memory in different ways.

File: project/visitorckw/README.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+- Enhance the user interface for ease of use.
+
+Week 7 (12/11): Testing and Debugging
+
+- Conduct extensive testing of the entire library, addressing any bugs and performance issues.
+- Optimize and refine the code for efficiency and correctness.
+
+Week 8 (12/18): Documentation and Finalization
+
+- Create comprehensive documentation, including a user guide and API reference.
```
Comment (yungyuc) [inline project/visitorckw/README.md:193]: It's probably very hard to create document in a week.  Writing useful documents may take more time than developing the code.

File: project/visitorckw/README.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+- Finalize the project by ensuring that all functionalities meet the project objectives.
+- Prepare for project submission and presentation.
```
Comment (yungyuc) [inline project/visitorckw/README.md:195]: One week is not sufficient for making an appealing presentation.

**Review by yungyuc — APPROVED** [review]
It's an executable plan.  I am not sure how impressive it is.  Please frequently update on github and the discord server.  If you come to the meetup this week https://sciwork.dev/meetup/2023/1108-nycu I may provide a little bit more help.

**Comment by yungyuc** [comment]
@visitorckw Please provide the github link to your project by making a global comment here or a new PR.

**Comment by yungyuc** [comment]
> @visitorckw Please provide the github link to your project by making a global comment here or a new PR.

@visitorckw ping

**Comment by visitorckw** [comment]
I apologize for not catching your message earlier.
Here's the link.
Link: https://github.com/visitorckw/runtime-profile-guided-optimization-sorting-library

**Pull request merged**